pkg/gadget-manifest/instance-spec.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gadgetconfig
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	Version          = 1
+	KindInstanceSpec = "instance-spec"
+)
+
+type InstanceSpec struct {
+	Version     int               `json:"version" yaml:"version"`
+	Kind        string            `json:"kind" yaml:"kind"`
+	ImageName   string            `json:"imageName" yaml:"imageName"`
+	ID          string            `json:"id" yaml:"id"`
+	Name        string            `json:"name" yaml:"name"`
+	Tags        []string          `json:"tags" yaml:"tags"`
+	ParamValues map[string]string `json:"paramValues" yaml:"paramValues"`
+}
```

Comment:
How should we validate this struct? Should we define a yaml schema?

Reply:
What exactly do you mean? Applying stricter rules to catch extra fields for example? Or to provide things like auto-completion in IDEs?

Currently, we check for `version` and `kind` when parsing, and for `image` before running.

Reply:
I meant to check things like:
- Name can't contain white spaces 
- Tags can't contain `,`
- etc.

Reply:
You're right, it makes sense to add those.